Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -827,12 +847,16 @@ void ContainerNode::removeChildren()
 
     Ref protectedThis { *this };
     NodeVector removedChildren;
-    removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren);
-
+    removeAllChildrenWithScriptAssertionMaybeAsync(ChildChange::Source::API, removedChildren, DeferChildrenChanged::No);
     rebuildSVGExtensionsElementsIfNecessary();
     dispatchSubtreeModifiedEvent();
 }
 
+void ContainerNode::delayDeletingRemovedChildren(NodeVector&& removedChildren, unsigned subTreeSize)
```

Comment:
I don't think this extra indirection is helpful. We should just directly call into async deletion queue in this function's call site instead.